AppInspector.RulesEngine/TextContainer.cs

```diff
@@ -383,7 +387,7 @@ public bool IsCommented(int index)
     /// <param name="scopes"> The scopes to check </param>
     /// <param name="boundary"> Boundary in the text </param>
     /// <returns> True if boundary is in a provided scope </returns>
-    public bool ScopeMatch(IEnumerable<PatternScope> scopes, Boundary boundary)
+    public bool ScopeMatch(IList<PatternScope> scopes, Boundary boundary)
```

Comment:
Curious about the reason for the change from IEnumerable to IList here

Reply:
It's more correct to what this intended to do. This api usually takes an array. But if someone were to provide an ienumerable that is not a list it would be enumerated multiple times in this method. We don't gain anything from the extra flexibility of enumerable so list serves the purpose here better. But IList allows either an array or a List.